### PR TITLE
BFD-2353: Gets rid of extra quotes in the FDA Value Code Names.

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-2353](https://jira.cms.gov/browse/BFD-2353)

**User Story or Bug Summary:**

Notes:

From AB2D

We had one of our partners escalate something they'd noticed in the results they pulled from our API, certain diagnostic DISPLAY_TEST values are coming with quotes in the value itself. It's consistent in at least codes A529, A530 and A539 and they've seen it in other codes as well. We checked our code and found this is how we're receiving it from you all, so wanted to make sure you were aware

AC

bfd-data-fda module includes the same functionality that exists within bfd-data-npi for sanitizing the raw values

---

### What Does This PR Do?

Replaces extra double quotes found in the drug lookup code names.

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

The Key store value for the fda drug code file hash needs to be cleared.

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public class FdaDrugCodeDisplayLookup {
    * Stores a map from Drug Code (PRODUCTNDC) to Drug Code Display (SUBSTANCENAME) derived from the
    * downloaded NDC file.
    */
-  private final Map<String, String> ndcProductHashMap = new HashMap<>();
+  private Map<String, String> ndcProductHashMap = new HashMap<>();
```
Comment (cbrunefearless) [inline apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:32]: Took out final so I can refactor the readFDACodeFile.

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java [hunk 2]
```diff
@@ -122,62 +122,88 @@ public String retrieveFDADrugCodeDisplay(Optional<String> claimDrugCode) {
...
+                      ndcProductColumns[1].indexOf("-") + 1, ndcProductColumns[1].length()),
+                  4,
+                  '0');
+          // ndcProductColumns[3] - Proprietary Name
+          // ndcProductColumns[13] - Substance Name
+          ndcProcessedData.put(
+              String.format("%s-%s", nationalDrugCodeManufacturer, nationalDrugCodeIngredient),
+              ndcProductColumns[3].replace("\"", "")
+                  + " - "
+                  + ndcProductColumns[13].replace("\"", ""));
```
Comment (lsmitchell) [inline apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:190]: Minor suggestion: Could be handy to just have these various magic indexes be static vars you can name descriptively.
`private static int PROPRIETARY_NAME_COLUMN_INDEX = 3`
etc

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java [hunk 3]
```diff
@@ -122,62 +122,88 @@ public String retrieveFDADrugCodeDisplay(Optional<String> claimDrugCode) {
...
+        } catch (StringIndexOutOfBoundsException e) {
+          continue;
```
Comment (lsmitchell) [inline apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:192]: Why do we ignore this exception?

File: apps/bfd-data-fda/src/test/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookupTest.java [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+  public void shouldReturnMappedFDADataWithoutExtraDoubleQuotes() throws IOException {
+
+    InputStream targetStream =
+        new ByteArrayInputStream(INPUT_FILE_STRING_WITH_DOUBLE_QUOTES.getBytes());
+
+    FdaDrugCodeDisplayLookup fdaDrugCodeDisplay =
+        FdaDrugCodeDisplayLookup.createDrugCodeLookupForTesting();
+
+    Map<String, String> results = fdaDrugCodeDisplay.getFdaProcessedData(true, targetStream);
+    assertEquals("Sterile Diluent - WATER", results.get("00000-0001"));
```
Comment (lsmitchell) [inline apps/bfd-data-fda/src/test/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookupTest.java:88]: That refactor made the test much easier to write!

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java (hunk 2 above)
Comment (cbrunefearless) [reply apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:190]: Added indexes for ProductNDC, PROPRIETARY_NAME and SUBSTANCENAME.

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java (hunk 3 above)
Comment (cbrunefearless) [reply apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:192]: took out the exception

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java [hunk 5]
```diff
@@ -123,61 +132,87 @@ public String retrieveFDADrugCodeDisplay(Optional<String> claimDrugCode) {
...
+                '0');
+        String nationalDrugCodeIngredient =
+            StringUtils.leftPad(
+                ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].substring(
+                    ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].indexOf("-") + 1,
+                    ndcProductColumns[PRODUCT_NDC_COLUMN_INDEX].length()),
+                4,
+                '0');
+        // ndcProductColumns[3] - Proprietary Name
+        // ndcProductColumns[13] - Substance Name
```
Comment (lsmitchell) [inline apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java:197]: Probably dont need this with our new index names!

**Review by lsmitchell — DISMISSED** [review]

File: apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java (hunk 5 above)
Comment (cbrunefearless) [reply apps/bfd-data-fda/src/main/java/gov/cms/bfd/data/fda/lookup/FdaDrugCodeDisplayLookup.java]: took out comments of fied names

**Review by mackec1 — APPROVED** [review]
LGTM

**Review by lsmitchell — APPROVED** [review]

**Pull request merged**